lib/embulk/filter/ruby_proc.rb

```diff
@@ -207,10 +235,22 @@ def add(page)
                   record_hash[col] = pr.call(record_hash[col], record_hash)
                 end
               end
-              page_builder.add(record_hash.values)
+              if page_procs.empty?
+                page_builder.add(record_hash.values)
+              else
+                proc_records << record_hash
+              end
             end
           end
         end
+
+        unless page_procs.empty?
+          tmp_records = page_procs.each_with_object([]) do |pr, arr|
+            result = pr.call(proc_records)
+            result.each { |r| arr << r }
+          end
+          tmp_records.each { |record| page_builder.add(record.values) }
```

Comment:
Do you have any reason why you wrote `each` loop twice?

Reply:
I removed an unnecessary loop. Thank you.